imageroot/actions/list-widget-options-for-UI/20readconfig

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python3
+
+#
+# Copyright (C) 2023 Nethesis S.r.l.
+# SPDX-License-Identifier: GPL-3.0-or-later
+#
+
+import json
+import os
+import sys
+import agent
+import subprocess
+
+rdb = agent.redis_connect() # full read-only access on every key
+modules=[]
+accepted_timezone_list = []
+list_of_timezone_widget = []
+
+# we query about all mail server to use it inside the user interface
+for key in agent.list_service_providers(rdb,'imap','tcp'):
```

Comment:
Why don't call directly the `list-service-providers` action from the UI?

Reply:
I followed the documentation 

https://nethserver.github.io/ns8-core/modules/service_providers/

```
import agent
rdb = agent.redis_connect(host='127.0.0.1') # connect the local Redis replica
print(agent.list_service_providers(rdb, 'imap', 'tcp'))
```

Reply:
however got roundcubemail we used module-uuid instead of module-uid to set the mail server, I do not know if it is important here cc @DavidePrincipi

Reply:
Every module has the `list-service-providers` action, there is no need to implement an action for use it, eg. :
```
[root@rl1 ~]# api-cli run module/webtop37/list-service-providers --data '{"service":"imap", "transport":"tcp"}'
Warning: using user "cluster" credentials from the environment
[{"port": "143", "host": "10.5.4.1", "node": "1", "user_domain": "webtop.test.net", "module_uuid": "9e1aa55b-efc8-49e8-8df6-253c247bd3d3", "mail_hostname": "rl1.leader.test.net", "transport": "tcp", "module_id": "mail1", "ui_name": null}]
```

Reply:
WebTop's `configure-module` action requires the module ID as input.

Reply:
you retrieve the uuid from uid

https://github.com/NethServer/ns8-webtop/blob/main/imageroot/actions/configure-module/20config#L199

Reply:
> Every module has the list-service-providers action, there is no need to implement an action for use it, eg. :
> [root@rl1 ~]# api-cli run module/webtop37/list-service-providers --data '{"service":"imap", "transport":"tcp"}'
> Warning: using user "cluster" credentials from the environment
> [{"port": "143", "host": "10.5.4.1", "node": "1", "user_domain": "webtop.test.net", "module_uuid": "9e1aa55befc8-49e8-8df6"

I tend to separate API from UI, API is there to give data to the UI, also maybe I am better in the API side :D

Reply:
Instead of calling `list-widget-options-for-UI', you have to call `list-service-providers` for the list of available mail modules and call a `list-accepted-timezones`  (that do only one thing)  to retrieve the list of acceptable timezones.

The same thing (data from API to UI) but with two separate and specific calls.